api.go

```diff
@@ -79,23 +79,28 @@ func handleNewAlert(a *App, w http.ResponseWriter, r *http.Request) (code int, m
 		alertData = alerttemplate.Data{}
 		n         = a.notifier
 	)
+	// decode request payload from Alertmanager in a struct
+	if err := json.NewDecoder(r.Body).Decode(&alertData); err != nil {
+		errMsg := fmt.Sprintf("Error while decoding alertmanager response: %s", err)
+		return http.StatusBadRequest, errMsg, nil, excepBadRequest, err
+	}
 	// fetch the room_name param. This room_name is used to map the webhook URL from the config.
 	// just an abstraction, for a more humanised version and to not end up making alertmanager config
 	// a mess by not flooding with google chat webhook URLs all over the place.
 	roomName := r.URL.Query().Get("room_name")
 	if roomName == "" {
-		return http.StatusBadRequest, "Missing required room_name param", nil, excepBadRequest, err
+		// try2
```

Comment:
Make the comment as `// Attempt to fetch the room name from the alert payload`.

Reply:
updated